File: toyProblem/sample.py
from pyspark import SparkContext
from pyspark.sql import SQLContext

app_name = "sample"
sc = SparkContext(appName = app_name)
sqlContext = SQLContext(sc)

df = sqlContext.createDataFrame([(1,2),(2,4),(3,8)],['a','b'])

sampleobj = sc._jvm.com.imshealth.Sample
sumValue = sampleobj.computeSum(df._jdf)

print(sumValue)

File: toyProblem/sample/Sample.scala
package com.imshealth

import org.apache.spark.sql.DataFrame
import org.apache.spark.{SparkContext}
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.functions._

object Sample{
    def computeSum(dataset: DataFrame): Long = {
        dataset.agg(sum("b")).first.getLong(0)
    } 
} 
